lib/index.js

```diff
@@ -117,17 +117,37 @@ class ConsulService {
           return resolve(null)
         }
 
-        const services = result.filter(service => {
-          let found = false
+        let services = []
 
-          service.ServiceTags.forEach(serviceTag => {
-            if (serviceTag === tag) {
-              found = true
-            }
+        // match a single tag
+        if (typeof tag === 'string') {
+          services = result.filter(service => {
+            let found = false
+
+            service.ServiceTags.forEach(serviceTag => {
+              if (serviceTag === tag) {
+                found = true
+              }
+            })
+
+            return found
           })
+        // match a set of tags
+        } else if (tag instanceof Array) {
+          services = result.filter(service => {
+            if (service.ServiceTags.length !== tag.length) {
+              return false
+            }
 
-          return found
-        })
+            for (let requiredTag of service.ServiceTags) {
+              if (tag.indexOf(requiredTag) === -1) {
+                return false
+              }
+            }
+
+            return true
+          })
+        }
 
```

Comment:
It might be nice to throw an error here if the type isn't supported so it doesn't slightly fail